src/main/java/edu/brown/cs/student/main/Main.java

```diff
@@ -73,33 +76,175 @@ private void run() {
       runSparkServer((int) options.valueOf("port"));
     }
 
-    // TODO: Add your REPL here!
-    MathBot math = new MathBot();
+    // initialize a StarDistance object
+    StarDistance distCalculator = new StarDistance();
+
     try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
       String input;
+      // read the INPUT by line
       while ((input = br.readLine()) != null) {
         try {
           input = input.trim();
-          String[] arguments = input.split(" ");
-          // TODO: complete your REPL by adding commands for addition "add" and subtraction
-          //  "subtract"
-          if (arguments.length == 3 && this.isNumeric(arguments[1]) && this.isNumeric(arguments[2])) { // && arguments[1] & arguments[2] are both numeric)
-            double num1 = Double.parseDouble(arguments[1]);
-            double num2 = Double.parseDouble(arguments[2]);
-            if (arguments[0].equalsIgnoreCase("add")){
-              System.out.println(math.add(num1, num2));
+          String[] args = input.split(" ");
```

Comment:
This also applies to my own code, but in the future to make code more readable, you could use more helper methods

Reply:
Yeah for this project I could definitely have used a lot more OOD and planned more when I just coded on the fly a gigantic REPL if else statement. Will pay attention to this next time and thanks for the feedback!